llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -183,7 +187,9 @@ static bool sinkScalarOperands(VPlan &Plan) {
         // TODO: Handle converting to uniform recipes as separate transform,
         // then cloning should be sufficient here.
         Instruction *I = SinkCandidate->getUnderlyingInstr();
-        Clone = new VPReplicateRecipe(I, SinkCandidate->operands(), true);
+        Clone = new VPReplicateRecipe(I, SinkCandidate->operands(), true,
+                                      /*Mask*/ nullptr,
+                                      *cast<VPReplicateRecipe>(SinkCandidate));
```

Comment:
nit: check `if (auto RepSinkCandidate = dyn_cast<...>)` above instead of `if (isa<...>)` and this `cast`?

Reply:
Done thanks!